### PR TITLE
Add basic sorting by tag for recipes

### PR DESCRIPTION
**Pull request by BogdanLinnik**

### CODE REVIEW

File: app/controllers/admin/recipes_controller.rb [hunk 1]
```diff
@@ -3,8 +3,17 @@ class Admin::RecipesController < ApplicationController
   before_action :get_required_data, only: [:new, :edit, :create]
 
   def index
-    @recipes = Recipe.paginate(page: params[:page],
-                               per_page: 10)
+    if params[:sort_by]
```
Comment (illia108) [inline app/controllers/admin/recipes_controller.rb]: would be good to extract this logic. Maybe, to the private method.

File: app/controllers/admin/recipes_controller.rb [hunk 2]
```diff
@@ -3,8 +3,17 @@ class Admin::RecipesController < ApplicationController
...
+      @sort_by = params[:sort_by].constantize
```
Comment (illia108) [inline app/controllers/admin/recipes_controller.rb]: No need to use instance variable. Just `sort_by`

File: app/controllers/admin/recipes_controller.rb [hunk 3]
```diff
@@ -3,8 +3,17 @@ class Admin::RecipesController < ApplicationController
...
+      unless @sort_by.find(params[:sort_value]).nil?
+        @filter = @sort_by.find(params[:sort_value])
```
Comment (illia108) [inline app/controllers/admin/recipes_controller.rb]: same here

File: app/views/admin/_recipe.html.erb [hunk 4]
```diff
@@ -49,7 +49,7 @@
 <% else %>
   <p>
     <% recipe.tags.each do |tag| %>
-      <%= link_to tag.name, admin_tag_path(tag) %> <!-- search by tag name will be implemented here -->
+      <%= link_to tag.name, admin_recipes_path(sort_by:'Tag', sort_value: tag.id)%> <!-- search by tag name will be implemented here -->
```
Comment (illia108) [inline app/views/admin/_recipe.html.erb]: As this feature will be implemented, please remove my comment

File: config/routes.rb [hunk 5]
```diff
@@ -5,6 +5,7 @@
   namespace :admin do
     resources :recipes do
       resources :comments
+      get 'sort_by_tag/:id', to: 'recipes#sort_by_tag', as: 'sort_by_tag'
```
Comment (illia108) [inline config/routes.rb]: Where this route is used? I didn't find `sort_by_tag` in recipes_controller

Comment (BogdanLinnik) [reply config/routes.rb]: my bad. I will remove it immedietly

File: app/controllers/admin/recipes_controller.rb (hunk 1 above)
Comment (BogdanLinnik) [reply app/controllers/admin/recipes_controller.rb]: ok

File: app/controllers/admin/recipes_controller.rb (hunk 2 above)
Comment (BogdanLinnik) [reply app/controllers/admin/recipes_controller.rb]: ok

File: app/controllers/admin/recipes_controller.rb [hunk 6]
```diff
@@ -65,6 +64,20 @@ def destroy
 
   private
 
+  def sort
+    if params[:sort_by]
+      @sort_by = params[:sort_by].constantize
+      unless @sort_by.find(params[:sort_value]).nil?
+        @filter = @sort_by.find(params[:sort_value])
+        @recipes = @filter.recipes.paginate(page: params[:page],
```
Comment (BogdanLinnik) [inline app/controllers/admin/recipes_controller.rb]: We need this instance variables for recipes index.html.erb.

Comment (illia108) [reply app/controllers/admin/recipes_controller.rb]: OK

**Pull request merged**